ompi/mca/coll/base/coll_base_allreduce.c

```diff
@@ -1247,133 +1247,109 @@ int ompi_coll_base_allreduce_intra_redscat_allgather(
     return err;
 }
 
-/**
- * A greedy algorithm to exchange data among processes in the communicator via
- * an allgather pattern, followed by a local reduction on each process. This
- * avoids the round trip in a rooted communication pattern, e.g. reduce on the
- * root and then broadcast to peers.
+/*
+ *   ompi_coll_base_allreduce_intra_allgather_reduce
  *
- * This algorithm supports both commutative and non-commutative MPI operations.
- * For non-commutative operations the reduction is applied to the data in the
- * same rank order, e.g. rank 0, rank 1, ... rank N, on each process.
+ *   Function:       use allgather for allreduce operation (for internode)
```

Comment:
Suggest omitting internode in the description because the algorithm can be used in other scenarios.